inc/saibridge.h

```diff
@@ -0,0 +1,376 @@
+     *
+     * @type sai_object_id_t
+     * @objects SAI_OBJECT_TYPE_BRIDGE
+     * @flags CREATE_AND_SET
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_BRIDGE_PORT_ATTR_BRIDGE_ID,
+
+    /**
+     * @brief FDB Learning mode
+     *
+     * @type sai_bridge_port_fdb_learning_mode_t
+     * @flags CREATE_AND_SET
+     * @default SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW
+     */
+    SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_MODE,
+
+    /**
+     * @brief Maximum number of learned MAC addresses
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_BRIDGE_PORT_ATTR_MAX_LEARNED_ADDRESSES,
+
+    /**
+     * @brief Action for packets with unknown source mac address
+     * when FDB learning limit is reached.
+     *
+     * @type sai_packet_action_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PACKET_ACTION_DROP
+     */
+    SAI_BRIDGE_PORT_ATTR_FDB_LEARNING_LIMIT_VIOLATION_PACKET_ACTION,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_BRIDGE_PORT_ATTR_END,
+
+    /** Custom range base value */
+    SAI_BRIDGE_PORT_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_BRIDGE_PORT_ATTR_CUSTOM_RANGE_END
+
+} sai_bridge_port_attr_t;
+
+
+/**
+ * @brief Create bridge port
+ *
+ * @param[out] bridge_port_id Bridge port ID
+ * @param[in] switch_id Switch object id
+ * @param[in] attr_count number of attributes
+ * @param[in] attr_list array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t(*sai_create_bridge_port_fn)(
+    _Out_ sai_object_id_t* bridge_port_id,
+    _In_ sai_object_id_t switch_id,
+    _In_ uint32_t attr_count,
+    _In_ const sai_attribute_t *attr_list);
+
+/**
+ * @brief Remove bridge port
+ *
+ * @param[in] bridge_port_id Bridge port ID
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t(*sai_remove_bridge_port_fn) (
+    _In_  sai_object_id_t bridge_port_id);
+
+/**
+ * @brief Set attribute for bridge port
+ *
+ * @param[in] bridge_port_id Bridge port ID
+ * @param[in] attr attribute to set
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t(*sai_set_bridge_port_attribute_fn)(
+    _In_ sai_object_id_t bridge_port_id,
+    _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Get attributes of bridge port
+ *
+ * @param[in] bridge_port_id Bridge port ID
+ * @param[in] attr_count number of attributes
+ * @param[inout] attr_list array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success Failure status code on error
+ */
+typedef sai_status_t(*sai_get_bridge_port_attribute_fn)(
+    _In_ sai_object_id_t bridge_port_id,
+    _In_ uint32_t attr_count,
+    _Inout_ sai_attribute_t *attr_list);
+
+/**
+ * @brief Attribute data for #SAI_BRIDGE_ATTR_TYPE
+ */
+typedef enum _sai_bridge_type_t
+{
+    /** vlan aware bridge */
+    SAI_BRIDGE_TYPE_1Q,
```

Comment:
in the current proposal, there is only one default 1Q bridge, so all 4k vlan objects are in the 1Q bridge. The all the vlan share same attribute such as max learned mac, and etc on that are defined in the bridge attribute.